test/runner/e2e_read_test.cpp

```diff
@@ -3,159 +3,69 @@
-}
-
-TEST_F(TinySnbReadTest, OptionalMatch) {
-    runTest(TestHelper::appendKuzuRootPath(
-        "test/test_files/tinysnb/optional_match/optional_match.test"));
-    runTest(
-        TestHelper::appendKuzuRootPath("test/test_files/tinysnb/optional_match/multi_label.test"));
-}
-
-TEST_F(TinySnbReadTest, OrderBy) {
-    runTestAndCheckOrder(
-        TestHelper::appendKuzuRootPath("test/test_files/tinysnb/order_by/single_label.test"));
-    runTestAndCheckOrder(
-        TestHelper::appendKuzuRootPath("test/test_files/tinysnb/order_by/multi_label.test"));
-}
-
-TEST_F(TinySnbReadTest, Union) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/union/union.test"));
-}
-
-TEST_F(TinySnbReadTest, Unwind) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/tinysnb/unwind/unwind.test"));
-}
-
-TEST_F(TinySnbReadTest, VarLengthExtendTests) {
-    runTest(TestHelper::appendKuzuRootPath(
-        "test/test_files/tinysnb/var_length_extend/var_length_adj_list_extend.test"));
-    runTest(TestHelper::appendKuzuRootPath(
-        "test/test_files/tinysnb/var_length_extend/var_length_column_extend.test"));
-}
-
-TEST_F(OneDimNpyReadTest, Match) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/npy-1d/match.test"));
-}
-
-class SingleSourceShortestPathTest : public DBTest {
-public:
-    std::string getInputDir() override {
-        return TestHelper::appendKuzuRootPath("dataset/shortest-path-tests/");
+    void TestBody() override {
+        for (auto& file : testFiles_) {
+            runTest(file);
+        }
     }
+
+private:
+    std::vector<std::string> testFiles_;
+    std::string dataset_;
 };
 
-TEST_F(SingleSourceShortestPathTest, BFS_SSSP) {
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/shortest_path/bfs_sssp.test"));
-    runTest(TestHelper::appendKuzuRootPath("test/test_files/shortest_path/bfs_sssp_large.test"));
+void RegisterE2ETests(const std::vector<TestConfig> testConfig) {
+    for (auto testItem : testConfig) {
+        testing::RegisterTest(testItem.testSuiteName.c_str(), testItem.testName.c_str(), nullptr,
+            testItem.testName.c_str(), __FILE__, __LINE__,
+            [=]() -> DBTest* { return new RunDynamicTest(testItem.files, testItem.dataset); });
+    }
 }
 
-class TinySnbSerialReadTest : public DBTest {
-public:
-    std::string getInputDir() override {
-        return TestHelper::appendKuzuRootPath("dataset/tinysnb-serial/");
+std::vector<TestConfig> scanEndToEndTestFiles(const std::string& path) {
+    TestConfig testConfig;
+    std::vector<TestConfig> tests;
+
+    for (const auto& entry : std::filesystem::directory_iterator(path)) {
+        std::string dirName = entry.path().filename().string();
+        testConfig.testSuiteName = TestHelper::convertSnakeCaseToCamelCase(dirName + "Test");
+        std::replace(dirName.begin(), dirName.end(), '_', '-');
+        testConfig.dataset = "dataset/" + dirName + "/";
```

Comment:
Can you take a look at `FileUtils::joinPath`? I think we can reuse that function to join path parts.